[ui5-tooling-transpile] Provide config option for build task to fail on d.ts file generation error

**Is your feature request related to a problem? Please describe.**
Right now, whenever there is an error in d.ts file generation like below:
```
error builder:custom-task:ui5-tooling-transpile-task The following errors occured during d.ts generation:
/resources/com/myorg/myapp/controller/Main.controller.ts(9,9): error TS4053: Return type of public method from exported class has or is using name 'SampleEnum' from external module "com/myorg/myapp/SampleEnum" but cannot be named.
```
, build process is not failing and thus `dist` with missing `d.ts` files may be silently generated, .e.g. during ci/cd pipeline.

**Describe the solution you'd like**
Configuration option of `ui5-tooling-transpile-task` to make `ui5 build` command fail in case d.ts generation has errors.

**Additional context**
Here is how to reproduce error descibed above during build process:
* Generate new project by using easy-ui5 generator and ts-app template
* Add `SampleEnum.ts` file to the `webapp` folder with following content:
```ts
enum SampleEnum {
	a = 'A',
	b = 'B'
}

export default SampleEnum;
```
* Add following method to `BaseController.ts`
```ts
	public getSample(): SampleEnum {
		return SampleEnum.a;
	}
```
* Add following method to `Main.controller.ts`
```ts
	public getFlattenedSample() {
		return this.getSample();
	}
```
* run `npm run build`

--- a/packages/ui5-tooling-transpile/README.md
+++ b/packages/ui5-tooling-transpile/README.md
@@ -47,6 +47,9 @@ npm install ui5-tooling-transpile --save-dev
 - generateDts: `boolean`
   if enabled, the tooling extension will generate type definitions (`.d.ts`) files; by default for projects of type `library` this option is considered as `true` and for other projects such as `application` this option is considered as `false` by default (is only relevant in case of transformTypeScript is `true`)
 
+- failOnDtsErrors: `boolean`
+  if enabled, the tooling extension will fail during the generation of type definitions (`.d.ts`) files; this option is considered as `false` by default (is only relevant in case of generateDts is `true`)
+
 - omitTSFromBuildResult: `boolean`
   if enabled, the TypeScript sources will be omitted from the build result. This will disable the debugging support in the TypeScript sources (since the related source files aren't included anymore)
 

--- a/packages/ui5-tooling-transpile/lib/task.js
+++ b/packages/ui5-tooling-transpile/lib/task.js
@@ -391,7 +391,7 @@ module.exports = async function ({ log, workspace /*, dependencies*/, taskUtil, 
 					await workspace.write(indexDtsFile);
 				} else {
 					// error diagnostics
-					log.error(
+					throw new Error(
 						`The following errors occured during d.ts generation: \n${ts.formatDiagnostics(
 							result.diagnostics,
 							host
@@ -400,7 +400,11 @@ module.exports = async function ({ log, workspace /*, dependencies*/, taskUtil, 
 				}
 			} catch (e) {
 				// typescript dependency should be available, otherwise we can't generate the dts files
+				// or if a d.ts file is not valid, the generation will fail and we report the error
 				log.error(`Generating d.ts failed! Reason: ${e.message}\n${e.stack}`);
+				if (config.failOnDtsErrors) {
+					throw e;
+				}
 			}
 		}
 	}

--- a/packages/ui5-tooling-transpile/lib/util.js
+++ b/packages/ui5-tooling-transpile/lib/util.js
@@ -333,6 +333,7 @@ module.exports = function (log) {
 				omitTSFromBuildResult: config.omitTSFromBuildResult,
 				generateTsInterfaces,
 				generateDts: config.generateDts,
+				failOnDtsErrors: config.failOnDtsErrors,
 				transpileDependencies: config.transpileDependencies,
 				transformAtStartup: config.transformAtStartup,
 				transformTypeScript,

